src/util/system.cpp

```diff
@@ -372,6 +373,86 @@ bool ArgsManager::IsArgSet(const std::string& strArg) const
     return !GetSetting(strArg).isNull();
 }
 
+static bool GetSettingsPath(fs::path* filepath = nullptr, bool temp = false)
+{
+    if (gArgs.IsArgNegated("-settings")) return false;
+    if (filepath) {
+        std::string settings = gArgs.GetArg("-settings", BITCOIN_SETTINGS_FILENAME);
+        *filepath = fs::absolute(temp ? settings + ".tmp" : settings, GetDataDir(/* net_specific= */ true));
+    }
+    return true;
+}
+
+bool ArgsManager::EnableSettingsFile()
+{
+    return GetSettingsPath();
+}
+
+bool ArgsManager::ReadSettingsFile()
+{
+    fsbridge::ifstream file;
+    fs::path filepath;
+    if (!GetSettingsPath(&filepath, /* temp= */ false)) return true;
+    file.open(filepath);
+    if (!file.is_open()) return true; // Ok for file not to exist.
+
+    util::SettingsValue in;
+    if (!in.read(std::string{std::istreambuf_iterator<char>(file), std::istreambuf_iterator<char>()})) {
+        LogPrintf("Error: Unable to parse settings file %s\n", filepath.string());
+        return false;
+    }
+    if (file.fail()) {
+        LogPrintf("Error reading settings file %s\n", filepath.string());
+        return false;
+    }
+    file.close();
+
+    if (!in.isObject()) {
+        LogPrintf("Error: Settings file %s is not in expected key-value format.\n", filepath.string());
+        return false;
+    }
+
+    LOCK(cs_args);
+    m_settings.rw_settings.clear();
+    const std::vector<std::string>& keys = in.getKeys();
+    const std::vector<UniValue>& values = in.getValues();
+    for (size_t i = 0; i < keys.size(); ++i) {
+        m_settings.rw_settings.emplace(keys[i], values[i]);
+    }
+    return true;
+}
+
+bool ArgsManager::WriteSettingsFile() const
```

Comment:
Without further PRs, this part is dead code,... would it at least make sense to add a test for it?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15935#discussion_r435744301

> Without further PRs, this part is dead code,... would it at least make sense to add a test for it?

Added a write call and tests